shell/common/shell.h

```diff
@@ -64,29 +156,89 @@ class Shell final : public PlatformView::Delegate,
+  ///             this method returns.
+  ///
   ~Shell();
 
+  //------------------------------------------------------------------------------
+  /// @return     The settings used to launch this shell.
+  ///
   const Settings& GetSettings() const;
 
+  //------------------------------------------------------------------------------
+  /// @brief      If callers wish to interact directly with any shell
+  ///             subcomponents. They must (on the platform thread) obtain a
+  ///             task runner that the component is designed to run on and a
+  ///             weak pointer to that component. They may then post a task to
+  ///             that task runner, do the validity check on that task runner
+  ///             before performing any operaation on that component. This
+  ///             accessor allows callers to acces the task runners for this
+  ///             shell.
+  ///
+  /// @return     The task runners current in use by the shell.
+  ///
   const TaskRunners& GetTaskRunners() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Rasterizers may only be accessed on the GPU task runner.
+  ///
+  /// @return     A weak pointer to the rasterizer.
+  ///
   fml::WeakPtr<Rasterizer> GetRasterizer();
 
+  //------------------------------------------------------------------------------
+  /// @brief      Engines may only be accessed on the UI thread.
+  ///
+  /// @return     A weak pointer to the engine.
+  ///
   fml::WeakPtr<Engine> GetEngine();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Platform views may only be accessed on the platform task
+  ///             runner.
+  ///
+  /// @return     A weak pointer to the platform view.
+  ///
   fml::WeakPtr<PlatformView> GetPlatformView();
 
-  DartVM* GetDartVM();
-
   // Embedders should call this under low memory conditions to free up
   // internal caches used.
   //
   // This method posts a task to the GPU threads to signal the Rasterizer to
   // free resources.
+
+  //----------------------------------------------------------------------------
+  /// @brief      Used by embedders to notify that there is a low memory
+  ///             warning. The shell will attempt to purge caches. Current, only
+  ///             the rasterizer cache is purged.
   void NotifyLowMemoryWarning() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Used by embedder to check if all shell subcomponents are
+  ///             initialized. It is embedder responsibility to make this call
+  ///             before accessing any other shell method. A shell that is not
+  ///             setup must be disarded and another one created with updated
+  ///             settings.
+  ///
+  /// @return     Returns if the shell has been setup. Once setup, this does not
```

Comment:
uber nit: set up (as opposed to setup the noun), unless I'm channelling my inner Canadian here.

Reply:
Done.